From: Marc Kleine-Budde <mkl@pengutronix.de>
Subject: Fix cross compilation

This patch fixes the cross compilation in another self writte makefiel

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>

---
 Makefile           |   13 +++++++------
 ubi-utils/Makefile |   14 +++++++-------
 2 files changed, 14 insertions(+), 13 deletions(-)

Index: mtd-utils-1.2.0/Makefile
===================================================================
--- mtd-utils-1.2.0.orig/Makefile
+++ mtd-utils-1.2.0/Makefile
@@ -1,9 +1,10 @@
 
 # -*- sh -*-
 
-SBINDIR=/usr/sbin
-MANDIR=/usr/share/man
-INCLUDEDIR=/usr/include
+PREFIX=/usr
+SBINDIR=${PREFIX}/sbin
+MANDIR=${PREFIX}/share/man
+INCLUDEDIR=${PREFIX}/include
 
 #CROSS=arm-linux-
 CC := $(CROSS)gcc
@@ -49,7 +50,7 @@ $(BUILDDIR)/%.o: %.c
 .SUFFIXES:
 
 all: $(TARGETS)
-	make -C $(BUILDDIR)/ubi-utils
+	$(MAKE) -C ubi-utils
 
 IGNORE=${wildcard $(BUILDDIR)/.*.c.dep}
 -include ${IGNORE}
@@ -57,7 +58,7 @@ IGNORE=${wildcard $(BUILDDIR)/.*.c.dep}
 clean:
 	rm -f $(BUILDDIR)/*.o $(TARGETS) $(BUILDDIR)/.*.c.dep $(SYMLINKS)
 	if [ "$(BUILDDIR)x" != ".x" ]; then rm -rf $(BUILDDIR); fi
-	make -C $(BUILDDIR)/ubi-utils clean
+	$(MAKE) -C ubi-utils clean
 
 $(SYMLINKS):
 	ln -sf ../fs/jffs2/$@ $@
@@ -104,4 +105,4 @@ install: ${TARGETS}
 	install -m0755 ${TARGETS} ${DESTDIR}/${SBINDIR}/
 	mkdir -p ${DESTDIR}/${MANDIR}/man1
 	gzip -9c mkfs.jffs2.1 > ${DESTDIR}/${MANDIR}/man1/mkfs.jffs2.1.gz
-	make -C $(BUILDDIR)/ubi-utils install
+	$(MAKE) -C ubi-utils install
Index: mtd-utils-1.2.0/ubi-utils/Makefile
===================================================================
--- mtd-utils-1.2.0.orig/ubi-utils/Makefile
+++ mtd-utils-1.2.0/ubi-utils/Makefile
@@ -3,10 +3,10 @@
 #
 
 KERNELHDR := ../include
-DESTDIR := /usr/local
-SBINDIR=/usr/sbin
-MANDIR=/usr/share/man
-INCLUDEDIR=/usr/include
+DESTDIR :=
+SBINDIR=$(PREFIX)/sbin
+MANDIR=$(PREFIX)/share/man
+INCLUDEDIR=$(PREFIX)/include
 
 CC := $(CROSS)gcc
 CFLAGS ?= -O2 -g -Werror
@@ -30,18 +30,18 @@ vpath   %.c ./src
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c -o $@ $< -g -Wp,-MD,.$(shell basename $<).dep
 
 all: $(TARGETS)
-	make -C new-utils
+	$(MAKE) -C new-utils
 
 IGNORE=${wildcard .*.c.dep}
 -include ${IGNORE}
 
 $(NTARGETS):
-	make -C new-utils $@
+	$(MAKE) -C new-utils $@
 	mv new-utils/$@ $@
 
 clean:
 	rm -rf *.o $(TARGETS) .*.c.dep
-	make -C new-utils clean
+	$(MAKE) -C new-utils clean
 
 pddcustomize: pddcustomize.o error.o libubimirror.o bootenv.o hashmap.o \
 		libubi.o crc32.o
